File: mvvm-architecture2/app/src/main/java/com/example/mvvmarchitecture2/MainActivity.java
package com.example.mvvmarchitecture2;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.BindingAdapter;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import com.example.mvvmarchitecture2.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
//        setContentView(R.layout.activity_main);
        ActivityMainBinding activityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        activityMainBinding.setViewM(new AppViewModel());
        activityMainBinding.executePendingBindings();
    }

    @BindingAdapter({"toastM"})
    public static void messageMonitored(View view, String message) {
        if (message != null) {
            Toast.makeText(view.getContext(), message, Toast.LENGTH_SHORT).show();
        }
    }
}

File: mvvm-todo-list-app-listadapter-with-viewbinding/app/src/main/java/com/example/mvvmtodolistappprojectsetup/ui/MainActivity.kt
package com.example.mvvmtodolistappprojectsetup.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import com.example.mvvmtodolistappprojectsetup.R
import com.example.mvvmtodolistappprojectsetup.databinding.ItemTaskBinding
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

    }
}